lib/adb.js

```diff
@@ -538,14 +538,38 @@ ADB.prototype.installApp = function installApp(deviceId, apkFile, opts, callback
 	} else {
 		this.devices(function (err, devices) {
 			if (err) return callback(err);
-			if (devices.filter(function (d) { return d.id == deviceId }).length != 1) return callback(new Error(__('device not found')));
+
+			devices = devices.filter(function (d) { return d.id == deviceId });
+			if (devices.length != 1) return callback(new Error(__('device not found')));
```

Comment:
In general, we prefer `!==` over `!=`. Same goes for the `==` on the previous line.